deployment/promotion/promote.py

```diff
@@ -227,110 +227,83 @@ def upload_artifact(auth, model_version_id, artifact):
     return put_url
 
 
-def upload_artifacts(auth, model_version_id, promoted_model):
-    print("Uploading %d artifacts" % len(promoted_model['artifact_keys']))
-    promoted_model['uploaded_artifacts'] = {}
+def upload_artifacts(auth, model_version_id, artifacts):
+    print("Uploading %d artifacts" % len(artifacts))
+    uploaded_artifacts = {}
 
-    for artifact in promoted_model['artifact_keys']:
-        promoted_model['uploaded_artifacts'][artifact["key"]] = upload_artifact(auth, model_version_id, artifact)
+    for artifact in artifacts:
+        uploaded_artifacts[artifact["key"]] = upload_artifact(auth, model_version_id, artifact)
+    return uploaded_artifacts
 
 
-def fetch_promotion_data(_config):
+def get_promotion_data(_config):
     source = _config['source']
-    
-    print("Fetching promotion data for '%s'" % source['endpoint_path'])
+    model_version_id = source['model_version_id']
+
+    print("Fetching promotion data for model version %d" % source['model_version_id'])
     source_auth = auth_context(source['host'], source['email'], source['devkey'], source['workspace'])
-    endpoint = get_endpoint(source_auth, source['endpoint_path'])
-    stages = get_endpoint_stages(source_auth, endpoint['id'])
+    model_version = get_model_version(source_auth, model_version_id)
+
+    all_builds = get_builds(source_auth, source)
 
-    promotion_data = {
-        'endpoint': endpoint,
-        'stages': []
+    time_format = '%Y-%m-%dT%H:%M:%S.%fZ'
+
+    model_version_builds = []
+    for build in all_builds['builds']:
+        if 'model_version_id' in build['creator_request'] and 'self_contained' in build['creator_request']:
+            if build['creator_request']['model_version_id'] == model_version_id:
```

Comment:
You already restricted by MV ID in the `get_builds` function, so why are you checking again

Reply:
I think you can get rid of this check and move the "no self contained builds found" check up above this loop, then handle just timestamps in this loop.

Reply:
You're right that the model version id check is redundant, I'll move that. But we won't know if there are any self contained builds or not until the loop is complete. Can also embed the date check into the loop as well